extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/sse/SseResource.java

```diff
@@ -0,0 +1,125 @@
+    public Multi<String> multiText() {
+        return Multi.createFrom().items("hello", "stef");
+    }
+
+    @Path("json")
+    @GET
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    @RestSseElementType(MediaType.APPLICATION_JSON)
+    public void sseJson(Sse sse, SseEventSink sink) throws IOException {
+        if (sink == null) {
+            throw new IllegalStateException("No client connected.");
+        }
+        SseBroadcaster sseBroadcaster = sse.newBroadcaster();
+
+        sseBroadcaster.register(sink);
+        sseBroadcaster.broadcast(sse.newEventBuilder().data(new Message("hello")).build())
+                .thenCompose(v -> sseBroadcaster.broadcast(sse.newEventBuilder().data(new Message("stef")).build()))
+                .thenAccept(v -> sseBroadcaster.close());
+    }
+
+    @Blocking
+    @Path("blocking/json")
+    @GET
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    @RestSseElementType(MediaType.APPLICATION_JSON)
+    public void blockingSseJson(Sse sse, SseEventSink sink) throws IOException {
+        if (sink == null) {
+            throw new IllegalStateException("No client connected.");
+        }
+        SseBroadcaster sseBroadcaster = sse.newBroadcaster();
+
+        sseBroadcaster.register(sink);
+        sseBroadcaster.broadcast(sse.newEventBuilder().data(new Message("hello")).build())
+                .thenCompose(v -> sseBroadcaster.broadcast(sse.newEventBuilder().data(new Message("stef")).build()))
+                .thenAccept(v -> sseBroadcaster.close());
+    }
+
+    @Path("json2")
+    @GET
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    public void sseJson2(Sse sse, SseEventSink sink) throws IOException {
+        if (sink == null) {
+            throw new IllegalStateException("No client connected.");
+        }
+        SseBroadcaster sseBroadcaster = sse.newBroadcaster();
+
+        // Same as sseJson but set mediaType in builder
+        sseBroadcaster.register(sink);
+        sseBroadcaster
+                .broadcast(sse.newEventBuilder().data(new Message("hello")).mediaType(MediaType.APPLICATION_JSON_TYPE).build())
+                .thenCompose(v -> sseBroadcaster.broadcast(
+                        sse.newEventBuilder().mediaType(MediaType.APPLICATION_JSON_TYPE).data(new Message("stef")).build()))
+                .thenAccept(v -> sseBroadcaster.close());
+    }
+
+    @Path("json/multi")
+    @GET
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    @RestSseElementType(MediaType.APPLICATION_JSON)
+    public Multi<Message> multiJson() {
+        return Multi.createFrom().items(new Message("hello"), new Message("stef"));
+    }
+
+    @Path("json/multi2")
+    @GET
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    public Multi<Message> multiDefaultElementType() {
+        return Multi.createFrom().items(new Message("hello"), new Message("stef"));
+    }
+
+    @Path("ndjson/multi")
+    @GET
+    @Produces(SseMediaType.APPLICATION_NDJSON)
+    @RestSseElementType(MediaType.APPLICATION_JSON)
```

Comment:
Now I wonder if we should not rename this `@RestStreamElementType`? Though to keep it compatible we'd keel `@RestSseElementType` and deprecate it in favour of the new one.

Reply:
I agree with you, however after taking a look to the required changes, there were many places to update within the resteasy reactive framework and I want to keep this PR the simpler the better. Wdyt?

Reply:
Well, I'd rather group this from the start, if we can, since we're going to document this.

Reply:
Fine with me to do this change. fyi @geoand

Reply:
I've done this in a separated commit: https://github.com/quarkusio/quarkus/pull/20908/commits/0c102bf51abd4c70e663ff042a05522d2d05335f